stripe/util.py

```diff
@@ -58,3 +72,38 @@ def utf8(value):
 def is_appengine_dev():
     return ('APPENGINE_RUNTIME' in os.environ and
             'Dev' in os.environ.get('SERVER_SOFTWARE', ''))
+
+
+def _console_log_level():
+    if stripe.log in ['debug', 'info']:
+        return stripe.log
+    elif STRIPE_LOG in ['debug', 'info']:
+        return STRIPE_LOG
+    else:
+        return None
+
+
+def log_debug(message, *args, **kwargs):
+    if _console_log_level() == 'debug':
+        print(message.format(*args, **kwargs))
+    logger.debug(message.format(*args, **kwargs))
+
+
+def log_info(message, *args, **kwargs):
+    if _console_log_level() in ['debug', 'info']:
+        print(message.format(*args, **kwargs))
+    logger.info(message.format(*args, **kwargs))
+
+def _test_or_live_environment():
+    if stripe.api_key is None:
+        return
+    match = re.match(r'sk_(live|test)_', stripe.api_key)
```

Comment:
Note that not all API keys follow the `sk_(live|test)_` convention. Some older keys simply start with `sk_` without the `(live|test)_` part, and some yet older keys don't have a prefix at all, they're just random alphanumeric characters.

Unfortunately for those keys the only way of knowing whether they're a test or live key would be to send an API request and check the `livemode` attribute in the response.